app/services/compute_resource_host_associator.rb

```diff
@@ -0,0 +1,36 @@
+class ComputeResourceHostAssociator
+  attr_accessor :compute_resource, :hosts, :fail_count
+  delegate :logger, :to => :Rails
+
+  def initialize(compute_resource)
```

Comment:
This class will generate a lot of sql queries. We should probably open a ticket to make this more efficient once this is merged.

Reply:
http://projects.theforeman.org/issues/22172